src/main/webapp/app/shared/dashboards/tutor-participation-graph/progress-bar/progress-bar.component.ts

```diff
@@ -1,51 +1,73 @@
-import { Component, Input, OnInit } from '@angular/core';
+import { ChangeDetectorRef, Component, Input, OnChanges, OnDestroy, OnInit, SimpleChanges } from '@angular/core';
 import { round } from 'app/shared/util/utils';
 import { Theme, ThemeService } from 'app/core/theme/theme.service';
+import { Subscription } from 'rxjs';
 
 @Component({
     selector: 'jhi-progress-bar',
     templateUrl: './progress-bar.component.html',
 })
-export class ProgressBarComponent implements OnInit {
+export class ProgressBarComponent implements OnInit, OnChanges, OnDestroy {
     @Input() public tooltip: string;
     @Input() public percentage: number;
     @Input() public numerator: number;
     @Input() public denominator: number;
 
-    constructor(private themeService: ThemeService) {}
+    foregroundColorClass: string;
+    backgroundColorClass: string;
+    themeSubscription: Subscription;
+
+    constructor(private themeService: ThemeService, private ref: ChangeDetectorRef) {}
 
     ngOnInit() {
-        this.percentage = round(this.percentage);
+        this.themeSubscription = this.themeService.getCurrentThemeObservable().subscribe(() => {
+            this.chooseProgressBarTextColor();
```

Comment:
Do you need this method call here? `this.ref.detectChanges();` calls `ngOnChanges()` which in turn calls `chooseProgressBarTextColor()`, right?

Reply:
If we run a change detection cycle, but there are no changes to be detected, then ``ngOnChanges`` would not be called. So, in this case, we first need to apply the change ( = choose a new progress bar css class name and store it in the component field ), then check for changes to be applied.

Reply:
I tested it without that line and it does indeed not work :D